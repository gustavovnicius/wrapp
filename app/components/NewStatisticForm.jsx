import React, {
  Component,
  PropTypes,
} from 'react';
import {
  Container,
  Input,
  Button,
} from 'rebass';

class NewStatisticForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: '',
      balance: '',
    };
  }

  handleNameChange(event) {
    this.setState({
      name: event.target.value,
    });
  }

  handleBalanceChange(event) {
    this.setState({
      balance: event.target.value,
    });
  }

  addStatistic() {
    this.props.onSubmit(this.state.name, parseFloat(this.state.balance));

    this.setState({
      name: '',
      balance: '',
    });
  }

  render() {
    return (
      <Container>
        <Input
          label=""
          name="name"
          placeholder="Name"
          value={this.state.name}
          onChange={event => this.handleNameChange(event)}
          rounded
          hideLabel
          type="text"
        />
        <Input
          label=""
          name="balance"
          placeholder="Balance"
          value={this.state.balance}
          onChange={event => this.handleBalanceChange(event)}
          rounded
          hideLabel
          type="number"
        />
        <Button style={{ width: '100%' }} onClick={() => this.addStatistic()}>
          Add
        </Button>
      </Container>
    );
  }
}

NewStatisticForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
};

export default NewStatisticForm;
