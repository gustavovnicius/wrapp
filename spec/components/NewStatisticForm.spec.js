import React from 'react';
import renderer from 'react-test-renderer';
import {
  shallow,
} from 'enzyme';
import {
  Input,
  Button,
} from 'rebass';
import NewStatisticForm from 'components/NewStatisticForm';

describe('NewStatisticForm', () => {
  it('match the snapshot', () => {
    const form = renderer.create(
      <NewStatisticForm onSubmit={jest.fn()} />,
    ).toJSON();

    expect(form).toMatchSnapshot();
  });

  describe('component events', () => {
    const name = 'Name';
    const balance = 1000;
    let spy;
    let form;

    beforeEach(() => {
      spy = jest.fn();
      form = shallow(
        <NewStatisticForm onSubmit={spy} />,
      );
    });

    it('call handleNameChange on name input change', () => {
      const event = {
        target: {
          value: name,
        },
      };

      form.find(Input).first().simulate('change', event);

      expect(form.instance().state.name).toEqual(name);
    });

    it('call handleBalanceChange on name input change', () => {
      const event = {
        target: {
          value: balance,
        },
      };

      form.find(Input).last().simulate('change', event);

      expect(form.instance().state.balance).toEqual(balance);
    });

    it('call addStatistic with name and balance state', () => {
      form.instance().setState({
        name,
        balance,
      });
      form.find(Button).simulate('click');

      expect(spy).toHaveBeenCalledWith(name, balance);
    });

    it('clear state after addStatistic is called', () => {
      form.instance().setState({
        name,
        balance,
      });
      form.find(Button).simulate('click');

      expect(form.instance().state).toEqual({ name: '', balance: '' });
    });
  });
});
