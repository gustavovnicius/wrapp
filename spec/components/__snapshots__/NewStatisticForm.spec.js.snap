// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewStatisticForm match the snapshot 1`] = `
<div
  className="Container"
  style={
    Object {
      "boxSizing": "border-box",
      "margin": "auto",
      "maxWidth": 1024,
      "paddingLeft": 16,
      "paddingRight": 16,
    }
  }
>
  <div
    className="Input"
    style={
      Object {
        "boxSizing": "border-box",
        "color": null,
        "marginBottom": 16,
      }
    }
  >
    <label
      className="Label"
      htmlFor="name"
      style={
        Object {
          "boxSizing": "border-box",
          "clip": "rect(1px, 1px, 1px, 1px)",
          "fontSize": 14,
          "fontWeight": 600,
          "height": 1,
          "lineHeight": 1,
          "overflow": "hidden",
          "position": "absolute",
          "width": 1,
        }
      }
    >
      
    </label>
    <input
      name="name"
      onChange={[Function]}
      placeholder="Name"
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, .25)",
          "borderColor": "rgba(0, 0, 0, .25)",
          "borderRadius": 2,
          "borderStyle": "solid",
          "borderWidth": 1,
          "boxSizing": "border-box",
          "color": "inherit",
          "display": "block",
          "fontFamily": "inherit",
          "fontSize": "inherit",
          "height": 32,
          "margin": 0,
          "paddingLeft": 8,
          "paddingRight": 8,
          "width": "100%",
        }
      }
      type="text"
      value=""
    />
  </div>
  <div
    className="Input"
    style={
      Object {
        "boxSizing": "border-box",
        "color": null,
        "marginBottom": 16,
      }
    }
  >
    <label
      className="Label"
      htmlFor="balance"
      style={
        Object {
          "boxSizing": "border-box",
          "clip": "rect(1px, 1px, 1px, 1px)",
          "fontSize": 14,
          "fontWeight": 600,
          "height": 1,
          "lineHeight": 1,
          "overflow": "hidden",
          "position": "absolute",
          "width": 1,
        }
      }
    >
      
    </label>
    <input
      name="balance"
      onChange={[Function]}
      placeholder="Balance"
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, .25)",
          "borderColor": "rgba(0, 0, 0, .25)",
          "borderRadius": 2,
          "borderStyle": "solid",
          "borderWidth": 1,
          "boxSizing": "border-box",
          "color": "inherit",
          "display": "block",
          "fontFamily": "inherit",
          "fontSize": "inherit",
          "height": 32,
          "margin": 0,
          "paddingLeft": 8,
          "paddingRight": 8,
          "width": "100%",
        }
      }
      type="number"
      value=""
    />
  </div>
  <button
    className="Button"
    href={undefined}
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#08e",
        "border": 0,
        "borderRadius": 2,
        "boxSizing": "border-box",
        "color": "#fff",
        "cursor": "pointer",
        "display": "inline-block",
        "fontFamily": "inherit",
        "fontSize": 14,
        "fontWeight": 600,
        "lineHeight": "16px",
        "margin": 0,
        "minHeight": 32,
        "paddingBottom": 8,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 8,
        "textDecoration": "none",
        "width": "100%",
      }
    }
  >
    Add
  </button>
</div>
`;
